### ISSUE TITLE
"oidc" configuration is not initialized when using THREESCALE_CONFIG_FILE

### ISSUE THREAD

**Issue by mayorova** [3scale/apicast#654]
When starting APIcast with `THREESCALE_PORTAL_ENDPOINT`, there is an extra step made in the loader which [initializes](https://github.com/3scale/apicast/blob/69ff05c03e934daf1fa3a9b2cfb4231773b5a22a/gateway/src/apicast/configuration_loader/remote_v2.lua#L180) and [configures](https://github.com/3scale/apicast/blob/69ff05c03e934daf1fa3a9b2cfb4231773b5a22a/gateway/src/apicast/configuration_loader/remote_v2.lua#L321-L325)  the `config.oidc` object, which is then merged with service configuration in the [configuration store](https://github.com/3scale/apicast/blob/69ff05c03e934daf1fa3a9b2cfb4231773b5a22a/gateway/src/apicast/configuration_store.lua#L106-L111).

When starting with `THREESCALE_CONFIG_FILE`, the [loader](https://github.com/3scale/apicast/blob/69ff05c03e934daf1fa3a9b2cfb4231773b5a22a/gateway/src/apicast/configuration_loader/file.lua) doesn't initialize `config.oidc`.

As a result, when `oidc` module [is loaded](https://github.com/3scale/apicast/blob/69ff05c03e934daf1fa3a9b2cfb4231773b5a22a/gateway/src/apicast/oauth/oidc.lua#L32-L47), if APIcast was started with `THREESCALE_CONFIG_FILE`, `service.oidc` will *only* have the `issuer_endpoint`, but not the complete config.

So, the JWT verification will fail. The first place where it will fail is the [alg check](https://github.com/3scale/apicast/blob/69ff05c03e934daf1fa3a9b2cfb4231773b5a22a/gateway/src/apicast/oauth/oidc.lua#L96), so APIcast logs will show:

```
2018/03/13 18:22:35 [debug] 91538#0: *5 [lua] proxy.lua:318: rewrite(): oauth failed with [jwt] invalid alg
```

Using `THREESCALE_CONFIG_FILE` is currently the only easy way to preview policies, and currently this can't be done with OIDC.

**Comment by mikz** [comment]
@nmasse-itix that is very useful, thank you!

**Comment by mikz** [comment]
This is going to be fixed in #850. Any input how it should behave is much welcomed. 

Right now it just loads any missing configuration.
